IdentityCore/src/oauth2/MSIDOauth2Factory.m

```diff
@@ -290,6 +290,12 @@ - (BOOL)fillRefreshToken:(MSIDRefreshToken *)token
         MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Trying to initialize refresh token when missing refresh token field");
         return NO;
     }
+    
+    // Nested auth protocol - RT is for 'broker/hub' client id
+    if (![NSString msidIsStringNilOrBlank:configuration.nestedAuthBrokerClientId])
```

Comment:
This should be moved to the AAD factory since this logic is AAD specific, and not OAuth2 specific

Reply:
Updated